core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt

```diff
@@ -1386,13 +1387,20 @@ abstract class CoreReaderFragment :
     externalLinkOpener?.openExternalUrl(intent)
   }
 
-  protected fun openZimFile(file: File, isCustomApp: Boolean = false) {
+  protected fun openZimFile(
+    file: File?,
+    isCustomApp: Boolean = false,
+    fileDescriptor: FileDescriptor? = null
+  ) {
```

Comment:
I am a bit disturb by this function signature.

I think it would be better to have two methods call `openZimFile` and `openZimFd` instead of having option arguments.

(But it is maybe because I'm not familiar with Kotlin language)

Reply:
@mgautierfr Here we only need to check if it is `file` or `fileDescriptor` that comes from the `CustomFileValidator` class and we are creating an `Archive` object accordingly. We will improve our code but first it should work with fileDescriptor.